library/types/src/modules/IP.rb

```diff
@@ -151,34 +141,25 @@ def ValidNetwork
     # @return ip address as integer
     def ToInteger(ip)
       return nil unless Check4(ip)
-
-      parts = ip.split(".")
-      parts.reduce(0) {|r, p| (r << 8) + p.to_i }
+      IPAddr.new(ip).to_i
     end
 
     # Convert IPv4 address from integer to string
     # @param [Fixnum] ip IPv4 address
     # @return ip address as string
     def ToString(ip)
-      parts = [16777216, 65536, 256, 1].map do |b|
-        (ip / b) & 255
-      end
-
-      parts.join(".")
+      IPAddr.new(ip, Socket::AF_INET).to_s
     end
 
     # Converts IPv4 address from string to hex format
     # @param [String] ip IPv4 address as string in "ipv4" format
     # @return [String] representing IP in Hex
-    # @example IP::ToHex("192.168.1.1") -> "0xC0A80101"
-    # @example IP::ToHex("10.10.0.1") -> "0x0A0A0001"
+    # @example IP::ToHex("192.168.1.1") -> "C0A80101"
+    # @example IP::ToHex("10.10.0.1") -> "0A0A0001"
```

Comment:
It hasn't, as you can see in the unchanged tests: https://github.com/yast/yast-yast2/blob/573553179d769ac5dcc08d37c05baeabfbc921c9/library/types/test/ip_test.rb#L146
Just the docs was wrong.